src/routing_client.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2015 MaidSafe.net limited.
+//
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+use sodiumoxide;
+use std::sync::mpsc;
+use std::thread::spawn;
+
+use id::Id;
+use action::Action;
+use event::Event;
+use messages::SignedToken;
+use routing_node::RoutingNode;
+use NameType;
+use data::{Data, DataRequest};
+use types::Bytes;
+use error::{RoutingError, ResponseError};
+use authority::Authority;
+use sodiumoxide::crypto;
+use messages::{ExternalRequest, ExternalResponse, InternalRequest, Content};
+
+type RoutingResult = Result<(), RoutingError>;
+
+/// Routing provides an actionable interface to RoutingNode.
+/// On constructing a new Routing object a RoutingNode will also be started.
+/// Routing objects are clonable for multithreading, or a Routing object can be
+/// cloned with a new set of keys while preserving a single RoutingNode.
+#[derive(Clone)]
+pub struct RoutingClient {
```

Comment:
I see why, but we are so close. Needs discussion as well. for now it will work though